### ISSUE TITLE
Pyright failed to understand pymysql's connection context manager.

### ISSUE THREAD

**Issue by dashmug** [microsoft/pyright#1374]
**Describe the bug**
Pyright complains about a class not implementing `__enter__` when it actually does.

**To Reproduce**
```python
# test.py
from pymysql import connect
from pymysql.connections import Connection

connection: Connection = connect(
    host="localhost",
    user="user",
    password="passwd",
    database="db",
    charset="utf8mb4",
)

with connection:
    with connection.cursor() as cursor:
        # Create a new record
        sql = "INSERT INTO `users` (`email`, `password`) VALUES (%s, %s)"
        cursor.execute(sql, ("webmaster@python.org", "very-secret"))

    with connection.cursor() as cursor:
        # Read a single record
        sql = "SELECT `id`, `password` FROM `users` WHERE `email`=%s"
        cursor.execute(sql, ("webmaster@python.org",))
        result = cursor.fetchone()
```

[PyMySQL's Connection class source code](https://github.com/PyMySQL/PyMySQL/blob/master/pymysql/connections.py#L92). You'll see it supports the Context Manager protocol (`__enter__` and `__exit__`).

Pyright complains about:

```
12:6 - error: Object of type "Connection" cannot be used with "with" because it does not implement __enter__ (reportGeneralTypeIssues)
```

**Expected behavior**
Pyright shouldn't complain as the class correctly implements the necessary methods. Mypy does not have the same problem. And the code runs as expected.

**VS Code extension or command-line**
I'm testing this on the command-line.

**Additional context**
Pyright version: 1.1.101
PyMySQL version: 1.0.2

**Comment by dashmug** [comment]
This may or may not be relevant. Context manager support was just [recently added to PyMySQL's Connection class](https://github.com/PyMySQL/PyMySQL/pull/886).

**Comment by erictraut** [comment]
Pyright is basing this decision on the typeshed pymsql type stubs. Based on the current stubs, pyright is correct in reporting this. It sounds like no one has updated the typeshed stubs since the change was made in the library. 

Pyright frequently pulls updates from typeshed, so if you file a bug in the typeshed repo — or even better, submit a PR, we can get the change incorporated relatively quickly in Pyright.

In the meantime, I'm going to close this bug since Pyright is doing the right thing here.
